### PR TITLE
Allow student and admin to change their password

### PR DESCRIPTION
**Pull request by davithss**
### Description
This PR creates a link on the top-right menu to change password then an edit page will be showed to update the password. The student and admins are allowed to use this link.

### Changelog
- create a devise registration controller
- customize devise edit view
- customize devise errors messages
- add on the top-right menu a link to edit password
- add a flash message on the internal layout
- add devise i18n for student and admin
- add devise registration routes for student and admin
- create registrations specs

### How to test
First, you must log in as a student or admin, secondly hit on your name on the top-right of the page and after clicking on change password link. 
On the change password screen, you need to fill it up with the following information: password, password confirmation, and your current password then click it on the save button.

### Screenshots

Top-right menu
image

Change Password View
image

Error message
image

### CODE REVIEW

File: app/controllers/registrations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class RegistrationsController < Devise::RegistrationsController
+  layout 'internal'
+
+  protected
+
+  def after_update_path_for(_resource)
```
Comment (basmoura) [inline app/controllers/registrations_controller.rb]: @davithss your param name is `_resource` but you are using the `resource`. I think that you don't need this parameter here

File: app/controllers/registrations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    case resource
+    when Admin
+      admin_dashboard_path
+    when Student
+      student_dashboard_path
+    end
+  end
```
Comment (basmoura) [inline app/controllers/registrations_controller.rb]: @davithss you can improve your code

```ruby
def after_update_path_for
   return admin_dashboard_path if resource.is_a?(Admin)
   student_dashboard_path
end
```

Comment (davithss) [reply app/controllers/registrations_controller.rb]: @basmoura Thank you for this suggestion but I think for this situation the case statement was easier to read and clear as well.

File: app/controllers/registrations_controller.rb (hunk 1 above)
Comment (davithss) [reply app/controllers/registrations_controller.rb]: Thanks for advising me. It's must be `#after_update_path_for(resource) as devise docs says.` without `_`.

**Pull request merged**